### PR TITLE
owner(ticdc):  fix changefeed state is automatically changed bug

### LINKED ISSUE #8330: Changefeed auto resumed after upstream pd error
1. pause a changefeed manually 
2. upstream unhealthy
3. ticdc ticdc changefeed and get the error 
```
[2023/02/14 06:11:12.346 +00:00] [ERROR] [changefeed.go:212] ["an error occurred in Owner"] [namespace=default] [changefeed=changefeed-870017] [error="[CDC:ErrReachMaxTry]reach maximum try: 10, error: rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster: rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster"]
```
4. ticdc reset changefeed state in handleError

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #8330

### What is changed and how it works?

When we handle the error of a stopped changefeed, the changefeed would be set to error state and retried by the owner. This behavior is unexpected. After this PR, TiCDC will not handle stopped changefeed errors and will just log them.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
Fix a bug that causes a stopped changefeed to restart unexpectedly.
```

### CODE REVIEW

**Review by nongfushanquan — COMMENTED** [review]
could you add a unit test to cover this case

File: cdc/owner/feed_state_manager.go [hunk 1]
```diff
@@ -432,6 +432,16 @@ func (m *feedStateManager) handleError(errs ...*model.RunningError) {
 		}
 	}
 
+	//  changefeed state from stopped to failed is allowed
+	// but stopped to error or normal is not allowed
+	if m.state.Info != nil && m.state.Info.State == model.StateStopped {
+		log.Warn("changefeed is stopped, do not change it's state",
```
[the remark this answers is not in the document]
Comment (asddongmen) [reply cdc/owner/feed_state_manager.go:438]:
```suggestion
		log.Warn("changefeed is stopped, ignore errors",
```

**Review by asddongmen — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**